### PR TITLE
doc: release notes for 12.2.8 luminous

### PR DESCRIPTION
**Pull request by theanalyst**
Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

### CODE REVIEW

**Comment by theanalyst** [comment]
jenkins render docs

File: doc/releases/luminous.rst [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+
+This is the eighth bug fix release of the Luminous v12.2.x long term stable
+release series. This release contains several bugfixes across all the components
+and we recommend all users upgrade.
+
+Upgrade Notes from previous luminous releases
+---------------------------------------------
+
+When upgrading from v12.2.5 or v12.2.6 please note that upgrade caveats from
+12.2.5 to 12.2.7 upgrades still apply to 12.2.8 as well.
```
Comment (liewegas) [inline doc/releases/luminous.rst]: apply to an upgrade from 12.2.5 to *any* version, including 12.2.8.

File: doc/releases/luminous.rst [hunk 2]
```diff
@@ -1,3 +1,109 @@
...
+luminious-12-2-5-upgrades_ .Upgrades from 12.2.7 or earlier releases like 12.2.4
+or not running bluestore can proceed normally. 12.2.8 has fixes for scrub to fix
+the warnings generated from 12.2.6 corruption of data digest.
```
Comment (liewegas) [inline doc/releases/luminous.rst]: Let's be more specific here.  New paragraph, and something like

In 12.2.7 we fixed the data digest regression from 12.2.6, but 12.2.7 clusters still generated health warnings like ```[ERR] 11.288 shard 207: soid
11:1155c332:::rbd_data.207dce238e1f29.0000000000000527:head data_digest
0xc8997a5b != data_digest 0x2ca15853
```.  12.2.8 improves the deep scrub code to automatically repair these inconsistencies.

**Comment by theanalyst** [comment]
jenkins render docs

File: doc/releases/luminous.rst [hunk 3]
```diff
@@ -1,3 +1,122 @@
...
+This is the eighth bug fix release of the Luminous v12.2.x long term stable
+release series. This release contains several bugfixes across all the components
+and we recommend all users upgrade.
+
+Upgrade Notes from previous luminous releases
+---------------------------------------------
+
+When upgrading from v12.2.5 or v12.2.6 please note that upgrade caveats from
+12.2.5 will apply to any _newer_ luminous version including 12.2.8. Please read
+the notes at luminious-12-2-5-upgrades_ .
```
Comment (smithfarm) [inline doc/releases/luminous.rst]: s/luminious/luminous/

File: doc/releases/luminous.rst [hunk 4]
```diff
@@ -1,3 +1,122 @@
...
+
+For the cluster that installed the broken 12.2.6 release, 12.2.7 fixed the
```
Comment (smithfarm) [inline doc/releases/luminous.rst]: s/For the cluster/For clusters/

**Comment by dzafman** [comment]
Abhishek,

The code for #23217 was a general fix in master for a very specific 
issue with user requested scrub repair.  It made sense to implement it 
first, then repair (fixed during deep scrub) the data digest distrust 
12.2.6 bug on top.

Luminous #23236 for 12.2.8 was the data digest distrust repair (fixed 
during deep scrub) which also included the backport of #23217.

The description of the #23217 backport portion is "An enhancement to 
scrub repair to handle data_digest_mismatch_info on replicas as long as 
all replicas' digests match each other."

David

**Comment by theanalyst** [comment]
@dzafman thanks  for the explanation, added a note, while the mail is already out, the release notes themselves and the blog posts themselves can be edited to add this info (along with a couple of rgw configurables which I missed in the announcement)

@cbodley @mattbenjamin can you review the rgw notable changes section

**Comment by theanalyst** [comment]
@liewegas final version hopefully :)

**Pull request merged**